pom.xml

```diff
@@ -276,6 +271,15 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.jenkins-ci.plugins</groupId>
```

Comment:
> Lots of other deps could probably be moved to dependencyManagement too; would be useful to have a tool which does this automatically based on source/bytecode scans.

I realize that the Jenkins community is fairly Maven-centric, but FYI Netflix's [Gradle Lint Plugin](https://github.com/nebula-plugins/gradle-lint-plugin) has this feature (c.f. [Unused Dependency Rule](https://github.com/nebula-plugins/gradle-lint-plugin/wiki/Unused-Dependency-Rule)).

Reply:
I think it is likely there is a Maven equivalent, I just have not looked for it yet.

Reply:
> I think it is likely there is a Maven equivalent, I just have not looked for it yet.

Looks like the [Maven Dependency Plugin](https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html) can list unused declared dependencies, though (just like the Gradle version) there seem to be too many false positives for it to be usable as part of any kind of automated enforcement.

Reply:
The problem of false positives is compounded by the heavy use of reflection, Groovy views, etc. etc. in Jenkins sources.